internal/db/resource.go

```diff
@@ -106,3 +104,68 @@ func (d *Database) ListResources(ctx context.Context, limit, offset int) (_ []Re
 	}
 	return resources, nil
 }
+
+// buildQuery is an utility function to build the database query according to two optional parameters.
+// nameFilter: used to match resources name.
```

Comment:
Can you please rename this to `namePrefixFilter` to make it clear this is not an exact match? Also, please include prefix search in the godoc.

Reply:
yes, thank you